pimcore/models/Object/Classificationstore.php

```diff
@@ -152,7 +152,10 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
     {
         $language  = $this->getLanguage($language);
 
-        if ($value) {
+        // treat value "0" nonempty
+        $nonEmpty = (is_string($value) || is_numeric($value)) && strlen($value) > 0;
+
+        if ($nonEmpty || $value) {
```

Comment:
Well, I haven't tested every possible type and whether those object's __toString() behave this way (returning always an empty string when empty/unset).

I tried to be consistent with the check in https://github.com/pimcore/pimcore/blob/4.3.0/pimcore/models/Object/ClassDefinition/Data.php#L190